server/plugin/graphql/lhs_request.go

```diff
@@ -22,72 +22,89 @@ const (
 func (c *Client) GetLHSData(ctx context.Context) ([]*github.Issue, []*github.Issue, []*github.Issue, error) {
 	orgsList := c.getOrganizations()
 	var resultReview, resultAssignee, resultOpenPR []*github.Issue
+
+	if len(orgsList) == 0 {
+		return c.fetchLHSData(ctx, resultReview, resultAssignee, resultOpenPR, false, "")
+	}
+
+	var err error
 	for _, org := range orgsList {
-		params := map[string]interface{}{
-			queryParamOpenPRQueryArg:    githubv4.String(fmt.Sprintf("author:%s is:pr is:%s archived:false", c.username, githubv4.PullRequestStateOpen)),
-			queryParamReviewPRQueryArg:  githubv4.String(fmt.Sprintf("review-requested:%s is:pr is:%s archived:false", c.username, githubv4.PullRequestStateOpen)),
-			queryParamAssigneeQueryArg:  githubv4.String(fmt.Sprintf("assignee:%s is:%s archived:false", c.username, githubv4.PullRequestStateOpen)),
-			queryParamReviewsCursor:     (*githubv4.String)(nil),
-			queryParamAssignmentsCursor: (*githubv4.String)(nil),
-			queryParamOpenPRsCursor:     (*githubv4.String)(nil),
+		resultReview, resultAssignee, resultOpenPR, err = c.fetchLHSData(ctx, resultReview, resultAssignee, resultOpenPR, true, org)
+		if err != nil {
+			return nil, nil, nil, err
 		}
+	}
 
+	return resultReview, resultAssignee, resultOpenPR, nil
+}
+
+func (c *Client) fetchLHSData(ctx context.Context, resultReview, resultAssignee, resultOpenPR []*github.Issue, setOrgCheck bool, org string) ([]*github.Issue, []*github.Issue, []*github.Issue, error) {
```

Comment:
Seems like the most frequent scenario is with an org, and that path always sets the `queryParamOpenPRQueryArg` etc twice.  Can we reverse it so the most common case is the most performant?

Reply:
Hey @wiggin77, I have implemented your suggestions and refactored the logic of adding the org check to the query.
Please have a look.

Reply:
@Kshitij-Katiyar this is an improvement in that the map is not recreated for each org.  However it does not address the issue I mentioned above. The three items that get overwritten in the case of `org != ""` should be reverse such that they are only overwritten when `org == ""`.

Reply:
(I'm assuming `org != ""` is the most frequent case.)

Reply:
@wiggin77  

Moving the org check directly into the params map definition and changing `if org != "" {` to `if org == "" {` would complicate handling cases where the org string is empty (possibly requiring regex or similar logic).  

An alternative would be to create two separate params maps—one with the org and one without—but that would increase memory usage.  

Given these trade-offs, the current implementation seems more efficient. Please let us know your thoughts.